ui/component/or-attribute-input/src/or-anomaly-config-chart.ts

```diff
@@ -0,0 +1,303 @@
+import { customElement, property } from "lit/decorators.js";
+import { html, PropertyValues} from "lit";
+import {Console, DefaultColor4, manager, Util} from "@openremote/core";
+import moment from "moment";
+import {throttle} from "lodash";
+import {AnnotationOptions} from "chartjs-plugin-annotation";
+import {
+    AnomalyDetectionConfigObject,
+    AnomalyDetectionConfiguration,
+    AnomalyDetectionConfigurationChange,
+    AnomalyDetectionConfigurationGlobal,
+    Asset,
+    AssetDatapointQueryUnion,
+    AssetModelUtil,
+    AssetQuery,
+    Attribute,
+    AttributeRef,
+    DatapointInterval,
+    ValueDatapoint
+} from "@openremote/model";
+import {i18next, translate} from "@openremote/or-translate";
+import "@openremote/or-components/or-collapsible-panel"
+import {GenericAxiosResponse, isAxiosError} from "@openremote/rest";
+
+
+
+@customElement("or-anomaly-config-chart")
+export class OrAnomalyConfigChart extends OrChart {
+
+    @property({type: Object})
+    public attributeRef?: AttributeRef = undefined;
+    @property({type: Object})
+    public anomalyConfig?: AnomalyDetectionConfiguration = undefined;
+    @property({type: Number})
+    public timespan?: Number = undefined;
+    @property({type: String})
+    private errorMessage = ""
+    @property({type: Boolean})
+    private canRefresh = false
+
+
+    protected willUpdate(changedProps: Map<string, any>) {
+        return super.willUpdate(changedProps);
+    }
+
+    protected async _loadData() {
+        let timespan=0
+        if(this._loading || !this.anomalyConfig || !this.canRefresh){
+            return
+        }
+        if(this.anomalyConfig.onOff == undefined || !this.anomalyConfig.type || !this.anomalyConfig.deviation){
+            return;
+        }
+        if(this.anomalyConfig.type==="global"){
+            if((this.anomalyConfig as AnomalyDetectionConfigurationChange).timespan === "" || !(this.anomalyConfig as AnomalyDetectionConfigurationGlobal).timespan
+                || !(this.anomalyConfig as AnomalyDetectionConfigurationGlobal).minimumDatapoints ||!/^P(\d+Y)?(\d+M)?(\d+W)?(\d+D)?(T(\d+H)?(\d+M)?(\d+S)?)?$/.test((this.anomalyConfig as AnomalyDetectionConfigurationGlobal).timespan as string)){
+                this.errorMessage = i18next.t("anomalyDetection.invalidConfiguration");
+                return
+            }
+        }else if(this.anomalyConfig.type==="change"){
+            if((this.anomalyConfig as AnomalyDetectionConfigurationChange).timespan === "" || !(this.anomalyConfig as AnomalyDetectionConfigurationChange).timespan
+                || !(this.anomalyConfig as AnomalyDetectionConfigurationChange).minimumDatapoints || !/^P(\d+Y)?(\d+M)?(\d+W)?(\d+D)?(T(\d+H)?(\d+M)?(\d+S)?)?$/.test((this.anomalyConfig as AnomalyDetectionConfigurationChange).timespan as string)){
+                this.errorMessage = i18next.t("anomalyDetection.invalidConfiguration");
+                return
+            }
+        }
```

Comment:
Seems like some duplicated code that can be merged into a single check that accepts both types.
Maybe move it to a separate function?